### ISSUE TITLE
Spurious error for variable declared and referenced in "$()" in PS1

### ISSUE THREAD

**Issue by lifecrisis** [koalaman/shellcheck#2098]
- Rule Id (if any, e.g. SC1000):

SC2154

- My shellcheck version (`shellcheck --version` or "online"):

ShellCheck - shell script analysis tool
version: 0.7.1
license: GNU General Public License, version 3
website: https://www.shellcheck.net

- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
PS1='$(x=1; echo $x)'
```

#### Here's what shellcheck currently says:

```
Line 2:
PS1='$(x=1; echo $x)'
^-- SC2154: x is referenced but not assigned.
```

#### Here's what I wanted or expected to see:

No error or warning.

**Comment by CleanCut** [comment]
I hit this as well today. This is an alias that shows you a pretty list of all branches in your local git repository that are not merged into your current branch.

```bash
alias gban='for k in $(git branch -a --no-merged | sed s/^..//);do echo -e $(git log --color -1 --pretty=format:"%Cgreen%ci %Cblue%cr%Creset" "$k")\\t"$k";done|sort'
```

![image](https://user-images.githubusercontent.com/5838512/110551050-076da800-80f2-11eb-8705-59f7d795fe31.png)

**Comment by TinCanTech** [comment]
```
#!/bin/sh
PS1='$(x=y; echo $x)'
```
https://www.shellcheck.net/
```
Line 2:
PS1='$(x=y; echo $x)'
^-- SC2154: x is referenced but not assigned.
```
**x** is assigned.
